app/controllers/application_controller.rb

```diff
@@ -2152,24 +2152,38 @@ def build_saved_audit(rec, eh)
     }
   end
 
-  def prov_redirect(typ=nil)
-    assert_privileges(params[:pressed])
-    if typ  # we need to do this check before doing anything to prevent
-            # history being updated
-      vm_ids = find_checked_items.map(&:to_i).uniq
-      if !typ.eql?("clone") && VmOrTemplate.includes_template?(vm_ids)
+  def task_supported?(typ)
+    vm_ids = find_checked_items.map(&:to_i).uniq
+    if %w(migrate publish).include?(typ) && VmOrTemplate.includes_template?(vm_ids)
+      render_flash_not_applicable_to_model(typ)
+      return
+    end
+
+    case typ
+    when "clone"
+      if vm_ids.present? && !VmOrTemplate.cloneable?(vm_ids)
         render_flash_not_applicable_to_model(typ)
         return
       end
-      if typ.eql?("clone") && vm_ids.present? && !VmOrTemplate.cloneable?(vm_ids)
+    when "migrate"
+      if vm_ids.present? && !VmOrTemplate.cloneable?(vm_ids)
```

Comment:
@chessbyte, seems like a change was lost during all the recent merge conflict/rebasing i had to do to this branch, pushing up the correct change shortly.